pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -148,6 +153,23 @@ public static URI getRetrieveAllSegmentWithTableTypeHttpUri(String host, int por
         rawTableName + TYPE_DELIMITER + tableType));
   }
 
+  public static URI getStartReplaceSegmentsURI(URI controllerURI, String rawTableName, String tableType)
+      throws URISyntaxException {
+    return new URI(StringUtil.join("/", StringUtils
```

Comment:
```suggestion
    return getURI(controllerURI.getScheme(), controllerURI.getHost(), controllerURI.getPort(), OLD_SEGMENT_PATH + '/' + rawTableName + START_REPLACE_SEGMENTS_PATH + TYPE_DELIMITER + tableType);
```

Reply:
`getURI()` encodes `?` to `%3F`.

```
http://localhost:18998/segments/mytable/startReplaceSegments%3Ftype=OFFLINE <- getURI(),
http://localhost:18998//segments/mytable/startReplaceSegments?type=OFFLINE  <- new URI(..)
```

When I use the request created by `getURI()`, I get 405 error.
```
org.apache.pinot.common.exception.HttpErrorStatusException: Got error status code: 405 (Method Not Allowed) with reason: "HTTP 405 Method Not Allowed" while sending request: http://localhost:18998/segments/mytable/startReplaceSegments%3Ftype=OFFLINE to controller: snlee-mn2.linkedin.biz, version: Unknown

```

Reply:
Maybe that's the reason why `getDeleteSegmentHttpUri()` and `getRetrieveAllSegmentWithTableTypeHttpUri()` are not using `getURI()`? We should fix it maybe, but out of the scope of this PR.